subprojects/execution/src/main/java/org/gradle/internal/execution/WorkExecutor.java

```diff
@@ -16,6 +16,6 @@
 
 package org.gradle.internal.execution;
 
-public interface WorkExecutor<R extends Result> {
-    R execute(UnitOfWork work);
+public interface WorkExecutor<C extends Context, R extends Result> {
```

Comment:
Can we get rid of the type parameters of `WorkExecutor`? I don't see us supporting different types in the future, so why carry around the burden of the type parameters?

Reply:
Yes, I think in the end we will be able to remove a bunch of type parameters. Again, I'd like to first move everything over, and then do a cleanup.

Reply:
I don't quite understand why you introduced the `Context` type parameter in the first place. We can do the cleanup afterwards.